openeogeotrellis/identity.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
```

Comment:
"identity" seems like a very generic name, while this is specifically about AWS user and token management?
If so, I think it's cleaner to move this somewhere under the "integrations" namespace, instead of making this a top-level module.

In the longer term I'd like to slim down the whole top-level of openeogeotrellis and make things more modular/isolated

Reply:
Identity is indeed quite generic but that was on purpose. At this stage we will indeed only use it for AWS like interactions (S3 access via S3 proxy) but there is no reason why this would not be applicable to any other type of services. If we have additional services that we build and deploy this identity can also be used to call those services where this identity will have assertions/claims. I don´t mind moving it to integrations module though. Perhaps naming it job_identity or job_identity_token given that it is really tied to a job execution?